reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md

```diff
@@ -0,0 +1,682 @@
+// feedback - the specific kind of feedback
+public bool CanAcceptFeedback(PredictionClient client, PredictorFeedbackKind feedback)
+{
+    return feedback switch
+    {
+        PredictorFeedbackKind.CommandLineAccepted => true,
+        _ => false,
+    };
+}
+
+// Get the predictive suggestions. It indicates the start of a suggestion rendering session.
+// client - Represents the client that initiates the call.
+// context - The PredictionContext object to be used for prediction.
+// cancellationToken - The cancellation token to cancel the prediction.
+public SuggestionPackage GetSuggestion(PredictionClient client, PredictionContext context,
+    CancellationToken cancellationToken)
+{
+    if (_candidates is not null)
+    {
+        string input = context.InputAst.Extent.Text;
+        List<PredictiveSuggestion>? result = null;
+
+        foreach (string c in _candidates)
+        {
+            if (c.StartsWith(input, StringComparison.OrdinalIgnoreCase))
+            {
+                result ??= new List<PredictiveSuggestion>(_candidates.Count);
+                result.Add(new PredictiveSuggestion(c));
+            }
+        }
+
+        if (result is not null)
+        {
+            return new SuggestionPackage(result);
+        }
+    }
+
+    return default;
+}
+
+// A command line was accepted to execute. The predictor can start processing early as needed
+// with the latest history.
+public void OnCommandLineAccepted(PredictionClient client, IReadOnlyList<string> history)
+{
+    // Reset the candidate state once the command is accepted.
+    _candidates = null;
+}
+
+public void OnSuggestionDisplayed(PredictionClient client, uint session, int countOrIndex) { }
+
+public void OnSuggestionAccepted(PredictionClient client, uint session, string acceptedSuggestion) { }
+
+public void OnCommandLineExecuted(PredictionClient client, string commandLine, bool success) { }
+
+#endregion;
+```
+
+## Step 7 - Build the feedback provider
+
+Now you are ready to build and begin using your feedback provider! To build the project, run the
+following command:
+
+```powershell
+dotnet build
+```
+
+This command create the PowerShell module as a DLL file in the following path of your project
+folder: `bin/Debug/net8.0/myFeedbackProvider`
+
+## Using a feedback provider
+
+To test your new feedback provider, import the compiled module into your PowerShell session. This
+can be done by importing the folder described after building has succeeded:
+
+```powershell
+Import-Module ./bin/Debug/net8.0/myFeedbackProvider
+```
+
+Once you're satisfied with your module, you should create a module manifest, publish it to the
+PowerShell Gallery, and install it in your `$env:PSModulePath`. For more information, see
+[How to create a module manifest][01]. You can add the `Import-Module` command to your `$PROFILE`
+script so the module is available in PowerShell session.
+
+You can get a list of installed feedback providers, using the following command:
```

Comment:
This requires the `PSSubsystemPluginModel` experimental feature to be enabled. But we should be good as the NOTE below explains it further.